compiler/src/dotty/tools/dotc/profile/JsonNameTransformer.scala

```diff
@@ -0,0 +1,46 @@
+package dotty.tools.dotc.profile
+
+import scala.annotation.internal.sharable
+
+// Based on NameTransformer but dedicated for JSON encoding rules
+object JsonNameTransformer {
+  private val nops = 128
+
+  @sharable private val op2code = new Array[String](nops)
+  private def enterOp(op: Char, code: String) = op2code(op.toInt) = code
+
+  enterOp('\"', "\\\"")
+  enterOp('\\', "\\\\")
+  // enterOp('/', "\\/") // optional, no need for escaping outside of html context
+  enterOp('\b', "\\b")
+  enterOp('\f', "\\f")
+  enterOp('\n', "\\n")
+  enterOp('\r', "\\r")
+  enterOp('\t', "\\t")
+
+  def encode(name: String): String = {
+    var buf: StringBuilder = null.asInstanceOf
+    val len = name.length
+    var i = 0
+    while (i < len) {
+      val c = name(i)
+      if (c < nops && (op2code(c.toInt) ne null)) {
+        if (buf eq null) {
+          buf = new StringBuilder()
+          buf.append(name.subSequence(0, i))
+        }
+        buf.append(op2code(c.toInt))
+      } else if (c <= 0x1F || c > 0x7F) {
```

Comment:
Just to check: is it indeed okay not to escape the DEL character?

I looked at [uJSON's `escapeByte`](https://github.com/com-lihaoyi/upickle/blob/fe53cf3b018aea5df232a71d4a49332781a91dab/upickle/core/src/upickle/core/RenderUtils.scala#L29) to try to get an answer, but it's not clear to me why DEL is excluded when `escapeUnicode` is true but included when it's false.

Reply:
I checked: the DEL character (U+007F) is non-printable, and both Node.js and Python escape it, even though they don’t escape all non-ASCII characters.

```
➜  ~ node                  
Welcome to Node.js v22.8.0.
Type ".help" for more information.
> JSON.stringify("\u007F")
'"\x7F"'
➜  ~ python3
Python 3.12.5 (main, Aug  6 2024, 19:08:49) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> json.dumps("\u007F")
'"\\u007f"'
```

Reply:
```suggestion
      } else if (c <= 0x1F || c >= 0x7F) {
```